dynamicPermission.SharedMailboxes.ReferenceExample.ps1

```diff
@@ -0,0 +1,735 @@
+            #     $SendonBehalfPermissionSplatParams = @{
+            #         Identity            = $permission.Name
+            #         GrantSendOnBehalfTo = @{add = "$($aRef.Guid)" }
+            #         Confirm             = $false
+            #     } 
+
+                    
+            #     if ($dryRun -eq $false) {
+            #         $addSendonBehalfPermission = Set-Mailbox @SendonBehalfPermissionSplatParams -ErrorAction Stop
+
+            #         $auditLogs.Add([PSCustomObject]@{
+            #                 # Action  = "" # Optional
+            #                 Message = "Successfully granted permission [SendonBehalf] to mailbox [$($permission.Value) ($($permission.Name))] for user [$($aRef.UserPrincipalName) ($($aRef.Guid))]"
+            #                 IsError = $false
+            #             })
+            #     }
+            #     else {
+            #         Write-Warning "DryRun: would grant permission [SendonBehalf] to mailbox [$($permission.Value) ($($permission.Name))] for user [$($aRef.UserPrincipalName) ($($aRef.Guid))]"
+            #     }
+            # }
+            # catch {
+            #     $ex = $PSItem
+            #     $errorMessage = Get-ErrorMessage -ErrorObject $ex
+                    
+            #     Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+            #     $auditLogs.Add([PSCustomObject]@{
+            #             # Action  = "" # Optional
+            #             Message = "Error granting permission [SendonBehalf] to mailbox [$($permission.Value) ($($permission.Name))] for user [$($aRef.UserPrincipalName) ($($aRef.Guid))]. Error Message: $($errorMessage.AuditErrorMessage)"
+            #             IsError = $True
+            #         })
+            # }
+        }
+    }
+
+    # Compare current with desired permissions and revoke permissions
+    $newCurrentPermissions = @{}
+    foreach ($permission in $currentPermissions.GetEnumerator()) {    
+        if (-Not $desiredPermissions.ContainsKey($permission.Name) -AND $permission.Name -ne "No Mailboxes Defined") {
+            # Revoke Exchange Online Mailbox permission
+            try {
+                Write-Verbose "Revoking permission [FullAccess] to mailbox [$($permission.Value) ($($permission.Name))] for user [$($aRef.UserPrincipalName) ($($aRef.Guid))]"
+
+                $FullAccessPermissionSplatParams = @{
+                    Identity        = $permission.Name
+                    User            = $aRef.Guid
+                    AccessRights    = 'FullAccess'
+                    InheritanceType = 'All'
+                    AutoMapping     = $AutoMapping
+                } 
+
+                if ($dryRun -eq $false) {
+                    $addFullAccessPermission = Add-MailboxPermission @FullAccessPermissionSplatParams -ErrorAction Stop
```

Comment:
Not tested. But Add-MailboxPermissions seems odd when revoking permissions

Reply:
corrected